### PR TITLE
8308236: Remove SystemDictionaryShared::clone_dumptime_tables()

### PR DESCRIPTION
**Pull request by iklam**
This function was necessary because we were patching the pointers inside `SystemDictionaryShared::_dumptime_table` when dumping the CDS archive. We no longer do that since [JDK-8307567](https://bugs.openjdk.org/browse/JDK-8307567), so this function and related code can be removed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8308236](https://bugs.openjdk.org/browse/JDK-8308236): Remove SystemDictionaryShared::clone_dumptime_tables()

### Reviewers
 * [Vladimir Ivanov](https://openjdk.org/census#vlivanov) (@iwanowww - **Reviewer**) ⚠️ Review applies to [7acaf92d](https://git.openjdk.org/jdk/pull/14020/files/7acaf92d2cb73a49b567b8b9e74fe0fcb6dcdd9d)
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14020/head:pull/14020` \
`$ git checkout pull/14020`

Update a local copy of the PR: \
`$ git checkout pull/14020` \
`$ git pull https://git.openjdk.org/jdk.git pull/14020/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14020`

View PR using the GUI difftool: \
`$ git pr show -t 14020`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14020.diff">https://git.openjdk.org/jdk/pull/14020.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14020#issuecomment-1550375522)

### CODE REVIEW

**Review by iwanowww — APPROVED** [review]
Looks good.

File: src/hotspot/share/cds/filemap.hpp [hunk 1]
```diff
@@ -473,7 +473,6 @@ class FileMapInfo : public CHeapObj<mtInternal> {
 
   static void allocate_shared_path_table(TRAPS);
   static void copy_shared_path_table(ClassLoaderData* loader_data, TRAPS);
-  static void clone_shared_path_table(TRAPS);
```
Comment (calvinccheung) [inline src/hotspot/share/cds/filemap.hpp:476]: I think the `copy_shared_path_table` could be removed as well.

**Review by calvinccheung — COMMENTED** [review]
Just one comment. Looks good otherwise.

File: src/hotspot/share/cds/filemap.hpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/cds/filemap.hpp]: I've removed it.

File: test/hotspot/jtreg/compiler/ciReplay/CiReplayBase.java [hunk 2]
```diff
@@ -155,8 +155,8 @@ public void cleanup() {
         removeFromCurrentDirectoryStartingWith("core");
         removeFromCurrentDirectoryStartingWith("replay");
         removeFromCurrentDirectoryStartingWith(HS_ERR_NAME);
-        remove(TEST_CORE_FILE_NAME);
-        remove(REPLAY_FILE_NAME);
+        remove(TEST_CORE_FILE_NAME);x
+        //remove(REPLAY_FILE_NAME);
```
Comment (calvinccheung) [inline test/hotspot/jtreg/compiler/ciReplay/CiReplayBase.java:159]: Change by accident?

Comment (iklam) [reply test/hotspot/jtreg/compiler/ciReplay/CiReplayBase.java]: Yes, I changed it by accident. I've reverted the change.

**Review by calvinccheung — APPROVED** [review]

**Comment by iklam** [comment]
Thanks @iwanowww and @calvinccheung for the review.
/integrate

**Pull request closed without merging**